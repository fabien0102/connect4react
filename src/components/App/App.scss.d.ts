export const app: string;
export const board: string;
export const column: string;