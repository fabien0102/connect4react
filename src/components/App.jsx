import React from 'react';
import styles from './App.css';

const App = () => (
  <div className={styles.app}>
    <h2>Hello fafa, how are you?</h2>
  </div>
);

export default App;