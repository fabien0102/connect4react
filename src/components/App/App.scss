@import '../../styles/variables.scss';

body {
  background: radial-gradient(ellipse at center, lightblue 0%, darken(lightblue, 60) 100%);
  color: white;
}

h1 {
  font-family: 'Raleway', serif;
}

.score {
  font-size: 2rem;
  height: 4rem;

}

$player-size: 4rem;
.yellow, .red {
  position: relative;
  top: 5px;
  font-size: $player-size;
  color: $yellow;

  transition: font-size 200ms ease-in-out;
  &.current {
    font-size: $player-size * 1.3;
  }
}

.red {
  color: $red;
}

.app {
  font-family: helvetica, arial, sans-serif;
  line-height: 200%;
  padding: 6px 20px 30px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.board {
  box-shadow: 10px 10px 50px rgba(black, 0.5);
  position: relative;
  display: flex;
  border-radius: 5px;
  border: 10px solid darken($background-color, 5);
  border-top: 10px solid lighten($background-color, 5);
  border-bottom: 10px solid lighten($background-color, 2);

  &:after {
    position: absolute;
    content: "";
    background-color: darken($background-color, 10);
    border-radius: 5px 5px 0 0;
    width: 30px;
    height: 200px;
    top: 100%;
    right: 0;
    transform: translate(100%, -50%);
  }

  &:before {
    position: absolute;
    content: "";
    background-color: darken($background-color, 10);
    border-radius: 5px 5px 0 0;
    width: 30px;
    height: 200px;
    top: 100%;
    left: 0;
    transform: translate(-100%, -50%);
  }
}

.column {
  display: flex;
  flex-direction: column-reverse;
  border-top: 10px solid $background-color;
  border-bottom: 10px solid $background-color;
}