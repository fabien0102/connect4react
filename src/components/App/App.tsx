import * as React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { Action } from 'redux-actions';
import * as ReactDOM from 'react-dom';
import * as styles from './App.scss';
import Cell from '../Cell/Cell';
import { addDisc, newGame, projectNextMove } from '../../actions/actions';
import { AppStore } from '../../store';
import { isNumber, get } from 'lodash';

interface AppProps {
  game: AppStore.Game;
  dispatch: Dispatch<any>;
}

export class App extends React.Component<AppProps, void> {
  render() {
    const { game, dispatch } = this.props;

    const board = game.board.map((column, columnIndex) => {
      const cells = column.map((cell, rowIndex) => {
        const isProjection = (columnIndex === get(game, 'nextCell.column') && rowIndex === get(game, 'nextCell.row'));
        return (
          <Cell
            key={`${columnIndex}-${rowIndex}`}
            isProjection={isProjection}
            column={columnIndex}
            value={isProjection ? game.currentPlayer : cell}
            onClick={col => dispatch(addDisc({ column: col }))}
            onMouseEnter={col => dispatch(projectNextMove({ column: col }))}
            />
        );
      });
      return (<div className={styles.column} key={columnIndex}>{cells}</div>);
    });

    let playerState = (<p>Current player: {game.currentPlayer}</p>);
    if (isNumber(game.winner) && game.winner !== 0) {
      playerState = (<h3>Player {game.winner} win! <button onClick={() => dispatch(newGame())}>New game</button></h3>);
    } else if (isNumber(game.winner) && game.winner === 0) {
      playerState = (<h3>Draw! <button onClick={() => dispatch(newGame())}>New game</button></h3>);
    }

    return (
      <div className={styles.app}>
        <h1>Connect 4 react</h1>
        {playerState}
        <div className={styles.board} onMouseLeave={() => dispatch(projectNextMove({ column: -1 }))}>{board}</div>
      </div>
    );
  }
};

const mapStateToProps = (state: any) => ({
  game: state.game
});

export default connect(mapStateToProps)(App);