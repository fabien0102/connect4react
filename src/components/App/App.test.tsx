import * as React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import 'jest';

describe('App', () => {
    it('should have correct text', () => {
        const app = shallow(<App counter={{ value: 0 }} dispatch={null} />);

        expect(app.find('h2').text()).toEqual('Hello fafa, how are you? Coucou Johnny');
    });
});