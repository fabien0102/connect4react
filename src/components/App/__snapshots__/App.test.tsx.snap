exports[`App should render correctly 1`] = `
<div
  className={undefined}>
  <h1>
    Connect 4 react
  </h1>
  <div
    className={undefined}>
    <i
      className="undefined undefined">
      ●
    </i>
    0
     - 
    0
    <i
      className="undefined ">
      ●
    </i>
  </div>
  <div
    className={undefined}
    onMouseLeave={[Function]}>
    <div
      className="undefined column-0">
      <div
        className="undefined undefined cell-0-0"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-0-1"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-0-2"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-0-3"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-0-4"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-0-5"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-0-6"
        onClick={[Function]}
        onMouseEnter={[Function]} />
    </div>
    <div
      className="undefined column-1">
      <div
        className="undefined undefined cell-1-0"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-1-1"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-1-2"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-1-3"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-1-4"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-1-5"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-1-6"
        onClick={[Function]}
        onMouseEnter={[Function]} />
    </div>
    <div
      className="undefined column-2">
      <div
        className="undefined undefined cell-2-0"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-2-1"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-2-2"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-2-3"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-2-4"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-2-5"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-2-6"
        onClick={[Function]}
        onMouseEnter={[Function]} />
    </div>
    <div
      className="undefined column-3">
      <div
        className="undefined undefined cell-3-0"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-3-1"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-3-2"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-3-3"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-3-4"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-3-5"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-3-6"
        onClick={[Function]}
        onMouseEnter={[Function]} />
    </div>
    <div
      className="undefined column-4">
      <div
        className="undefined undefined cell-4-0"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-4-1"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-4-2"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-4-3"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-4-4"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-4-5"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-4-6"
        onClick={[Function]}
        onMouseEnter={[Function]} />
    </div>
    <div
      className="undefined column-5">
      <div
        className="undefined undefined cell-5-0"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-5-1"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-5-2"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-5-3"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-5-4"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-5-5"
        onClick={[Function]}
        onMouseEnter={[Function]} />
      <div
        className="undefined undefined cell-5-6"
        onClick={[Function]}
        onMouseEnter={[Function]} />
    </div>
  </div>
  
  
</div>
`;
